### DISCUSSION TITLE
Using a shared remote db for development

### DISCUSSION THREAD

**Discussion by thupsi** [craftcms/cms#15462] [Q&A] [answered]
One of the teams I'm working with is mainly a Wordpress shop, but I've been constantly preaching about Craft and they are willing to give it a shot. Hooray! 🎉

Recently, the team has been experimenting with using a shared remote db for Wordpress development and they suggested trying the same with Craft. Having had no experience with doing this with Craft, I'm a little skeptical:

- Is it viable at all?
- Is it recommended?
- What would be the big gotchas in this case?
- What are the actual steps to connect to a remote db with Craft? (Using DDEV)

Thanks in advance for any help on this!

**Comment by brandonkelly** [comment, marked as answer]
It’s been done, but we don’t recommend it:

- The performance is terrible. It adds a significant lag to each database query, so expect to spend a lot of time waiting around for page loads.
- Project config YAML will get pretty messy, because any time it’s generated, you’re going to end up seeing more changes than you were expecting, since it will also include whatever changes other developers have made (likely for things they haven’t even committed yet). You‘d probably end up just [opting out](https://craftcms.com/docs/5.x/system/project-config.html#manual-yaml-file-generation) of automatic project config YAML generation, and forgoing the entire project config workflow.
- It wouldn’t be possible to work on features in their own branches, since everyone is sharing the same database.

I would strongly encourage them to use individual dev environments, and embrace using project config to keep things in sync.

**Reply by thupsi** [reply]
Thanks @brandonkelly! I was thinking more or less along those same points, but I much prefer having an official and more informed answer!
